CONTRIBUTING.md

```diff
@@ -0,0 +1,138 @@
+# Contributing
+
+One of the best ways to contribute is by writing templates!
+
+### What are Templates?
+
+A template is a piece of code written in a templating language called
+[Jinja](https://jinja.palletsprojects.com/en/3.0.x/). A template defines
+a function that maps an example from a dataset in the
+[HuggingFace library](https://huggingface.co/datasets) to two strings of
+text. The first is called the _prompt_ which provides all information that
+will be available to solve a task, such as the instruction and the context.
+The second piece is called the _output_, which is the desired response to the
+prompt.
+
+### Quick-Start Guide to Writing Templates
+
+1. **Set up the app.** Fork the app and set up using the
+[README](https://github.com/bigscience-workshop/promptsource/blob/main/README.md).
+1. **Select a dataset.** Go to the tracking spreadsheet
+[here](https://docs.google.com/spreadsheets/d/10SBt96nXutB49H52PV2Lvne7F1NvVr_WZLXD8_Z0JMw/)
+and find an unclaimed one. Put your name under "Who's Prompting it?" and
+mark it yellow to show it's in progress.
+1. **Examine the dataset.** Select the dataset from the dropdown in the app.
+If the dataset has subsets (subsets are not the same as splits), you can select
+which one to work on. Note that templates are subset specific. You can find
+out background information on the dataset by reading the information in the
+app. The dataset is a collection of examples, and each example is a Python
+dictionary. The sidebar will tell you the schema that each example has.
+1. **Start a new template**. Enter a name for your first template and hit "Create."
+You can always update the name later. If you want to cancel the template, select
+"Delete Template."
+1. **Write the template**. In the box labeled "Template," enter a Jinja expression.
+See the [getting started guide](#getting-started-using-jinja-to-write-templates)
+and [cookbook](#jinja-cookbook) for details on how to write templates.
+1. **Add a reference.** If your template was inspired by a paper, note the
```

Comment:
What should the reference format be? Copy-paste bibtext from google scholar?

Reply:
I'm not picky. I think Google Scholar bibtex is such a mess usually, I was just formatting in a human readable way, like
```
Making Pre-trained Language Models Better Few-shot Learners. Tianyu Gao, Adam Fisch, and Danqi Chen. ArXiv 2020.
```
But if wanted to create an automatic bibliography, that could be cool. In that case, we might need a bibtex specific field as well because we're already mixing notes and things in the "reference."

A solution could be: only bibtex in reference, any other description should go in the name?

Reply:
Ehh... I'm already cooling on this idea. I would prefer to just leave it unconstrained unless someone feels differently.

Reply:
I don't have a strong opinion on this
as long as it's easy to find the reference, it should do the trick
at least: link to pdf + table number or link to pdf + permalink to the line of code